### PR TITLE
🧪 Add tests for useDefaultPlaylistItems hook

### PR DESCRIPTION
**Pull request by is0692vs**
What: Added unit tests for the useDefaultPlaylistItems hook.
Why: To improve test coverage and ensure the hook correctly handles fetching the default playlist, mapping the response, caching to local storage, and edge cases like missing session or network errors.
Verification: Verified by running the specific test file and the full test suite.
Result: The hook is now fully covered by tests.

---
*PR created automatically by Jules for task [13982604712924938405](https://jules.google.com/task/13982604712924938405) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

`useDefaultPlaylistItems` フックに対するユニットテストを新規追加したPRです。成功時のデータ変換・キャッシュ、空の `items`、未認証、ネットワークエラー、非OKレスポンスの5ケースをカバーしており、ロジック的な誤りはありません。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

テストのみの変更でありプロダクションコードへの影響はなく、マージ安全です。

P1以上の問題は見当たらず、残るフィードバックはすべてP2（スタイル・堅牢性の改善）です。テストカバレッジは主要ケースを網羅しており、ロジックの誤りもありません。

特に注意が必要なファイルはありません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx | useDefaultPlaylistItems フックの新規テストファイル。5ケースをカバーし、ロジック自体は正確。軽微なスタイル上の改善余地あり（clearAllMocks/restoreAllMocks の重複、wrapper のスコープ）。 |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Test as テストケース
    participant Hook as useDefaultPlaylistItems
    participant RQ as React Query
    participant Fetch as global.fetch
    participant Cache as setArticlesCache

    Test->>Hook: renderHook()
    Hook->>RQ: useQuery({ enabled: !!userEmail })
    alt userEmail あり
        RQ->>Fetch: fetch("/api/playlists/default")
        alt レスポンス ok
            Fetch-->>RQ: { id, name, items }
            RQ-->>Hook: isSuccess = true
            Hook->>Cache: setArticlesCache(email, data)
        else レスポンス not ok
            Fetch-->>RQ: { ok: false }
            RQ-->>Hook: isError = true
        else ネットワークエラー
            Fetch-->>RQ: throw Error
            RQ-->>Hook: isError = true
        end
    else userEmail なし
        RQ-->>Hook: isPending=true, fetchStatus=idle
    end
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx
Line: 41-43

Comment:
**`clearAllMocks` と `restoreAllMocks` の重複**

`beforeEach` で `jest.clearAllMocks()`、`afterEach` で `jest.restoreAllMocks()` を両方使っていますが、このテストでは `jest.spyOn` を一切使っていないため、`restoreAllMocks` はほぼ無効です。`clearAllMocks` だけで十分で、`afterEach` ブロックは削除するか `clearAllMocks` に統一する方がシンプルです。

```suggestion
    afterEach(() => {
        jest.clearAllMocks();
    });
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx
Line: 76-84

Comment:
**`setArticlesCache` の検証を `waitFor` でラップすることを推奨**

`useEffect` 内の `setArticlesCache` 呼び出しはクエリ成功後のレンダーの後に非同期で実行されます。`waitFor(() => expect(result.current.isSuccess).toBe(true))` が解決した時点では `act()` がエフェクトをフラッシュするため通常は問題ありませんが、テストの安定性を高めるために `setArticlesCache` の検証自体も `waitFor` でラップするとより明示的です。

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx
Line: 45-47

Comment:
**`wrapper` をテストごとに再生成することを推奨**

`wrapper` は `describe` スコープで定義された `const` ですが、内部で参照している `queryClient` は `let` 変数です。クロージャ経由で正しく動作しますが、意図がわかりにくくなります。明示性を高めるため、`beforeEach` 内で `wrapper` を再生成するか、各テストに渡すパターンが一般的です。

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test: add tests for useDefaultPlaylistIt..."](https://github.com/hiroki-org/audicle/commit/ef6d464c93f904a3c25cea57c5bd65552dcb3c97) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=29095557)</sub>

> Greptile also left **3 inline comments** on this PR.

### CODE REVIEW

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        // Setup default mocks
+        const { useSession } = require("next-auth/react");
+        useSession.mockReturnValue(mockSession);
+
+        global.fetch = jest.fn();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        // Verify caching was triggered
+        expect(setArticlesCache).toHaveBeenCalledWith(
+            "test@example.com",
+            {
+                playlistId: "playlist-1",
+                playlistName: "Default Playlist",
+                items: mockResponse.items,
+            }
+        );
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        global.fetch = jest.fn();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+    );
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import { renderHook, waitFor } from "@testing-library/react";
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                queries: {
+                    retry: false, // Disable retries for faster tests
+                },
+            },
+        });
+        jest.clearAllMocks();
+
+        // Setup default mocks
+        const { useSession } = require("next-auth/react");
+        useSession.mockReturnValue(mockSession);
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+        global.fetch = jest.fn();
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。

File: packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        expect(result.current.data).toEqual({
+            playlistId: "playlist-2",
+            playlistName: "Empty Playlist",
+            items: [],
+        });
+    });
+
+    it("should not fetch when userEmail is undefined", async () => {
+        const { useSession } = require("next-auth/react");
+        useSession.mockReturnValue({ data: null, status: "unauthenticated" });
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/lib/hooks/__tests__/useDefaultPlaylistItems.test.tsx]: 確認ありがとうございます。この指摘は commit aff92be で対応済みのため、会話をresolveします。
